src/network/networkpacket.h

```diff
@@ -118,6 +119,32 @@ class NetworkPacket
 	NetworkPacket &operator>>(video::SColor &dst);
 	NetworkPacket &operator<<(video::SColor src);
 
+	template<typename T> NetworkPacket &operator >> (std::vector<T> &dst)
+	{
+		u16 count;
+		*this >> count;
+
+		while (count-- > 0) {
```

Comment:
is underflow possible if count == 0 ?

Reply:
No, `count--` ensures that the value before decreasing is checked against `> 0`.